### PR TITLE
EVEREST-482 Removed nolint:funlen,cyclop directives

### PR DESCRIPTION
**Pull request by gen1us2k**
[EVEREST-482](https://jira.percona.com/browse/EVEREST-482) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**CHANGE DESCRIPTION**
---
**Problem:**
EVEREST-482

*Short explanation of the problem.*
We shut some linters intentionally in the past 

**Cause:**
*Short explanation of the root cause of the issue if applicable.*

**Solution:**
*Short explanation of the solution we are providing with this PR.*

The linter complains about cyclomatic complexity of 11 or 12 and I didn't see any other option instead of increasing this to 15. 15 is good enough and I had the setting for 30 in the past. Having the default setting for funlen and max-complexity set to 15 will prevent us from writing huge functions.

I refactored functions that had funlen issues but the rest of functions with `//nolint:cyclop` they had a cyclomatic complexity of 11 or 12 and there's no sane way to refactor them 

**CHECKLIST**
---
**Jira**
- [ ] Is the Jira ticket created and referenced properly?

**Tests**
- [ ] Is an Integration test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -2,6 +2,9 @@
 # Almost all linters; some of them are optional.
 
 linters-settings:
+  cyclop:
+    max-complexity: 15
```
Comment (recharte) [inline .golangci.yml:6]: This seems like a cheat :see_no_evil:

**Comment by recharte** [comment]
Can we start being strict with the PR description from now on? I'm seeing many PRs being opened with the default template.
I know that this particular PR doesn't add any new functionality or fix any bug but if we slack on these PRs we'll start slacking on any PR. Let's do a team effort to fill the template correctly for every PR.

**Comment by gen1us2k** [comment]
>Can we start being strict with the PR description from now on? I'm seeing many PRs being opened with the default template.
> I know that this particular PR doesn't add any new functionality or fix any bug but if we slack on these PRs we'll start slacking on any PR. Let's do a team effort to fill the template correctly for every PR.

I agree. I was in the middle of filling it

File: .golangci.yml (hunk 1 above)
Comment (recharte) [reply .golangci.yml:6]: Do you think that by quoting Linkin Park I'll become so numb that I can't feel you there? It seems like you're holding too tightly, afraid to lose control... :see_no_evil: :joy:

Jokes aside... Maybe we can actually split some of these further, I can try to take a look as well...

**Review by oksana-grishchenko — APPROVED** [review]
I thinks it's still ok to have 15 as max complexity. In case of API method handlers we need to do many different things in it and sometimes splitting the method into smaller brings more confusion than if we have all logic plain.

File: api/monitoring_instance.go [hunk 2]
```diff
@@ -180,29 +180,36 @@ func (e *EverestServer) DeleteMonitoringInstance(ctx echo.Context, name string)
...
+	if err != nil {
+		return ctx.JSON(http.StatusInternalServerError, Error{
+			Message: pointer.ToString(err.Error()),
+		})
+	}
+
+	return ctx.NoContent(http.StatusNoContent)
+}
+
+func (e *EverestServer) deleteMonitoringConfig(c context.Context, i *model.MonitoringInstance) error {
```
Comment (michal-kralik) [inline api/monitoring_instance.go:199]:
```suggestion
func (e *EverestServer) deleteMonitoringConfig(ctx context.Context, i *model.MonitoringInstance) error {
```

**Review by michal-kralik — APPROVED** [review]

**Pull request merged**